tx_test.go

```diff
@@ -70,6 +71,28 @@ func TestTx_Transaction(t *testing.T) {
 		var dest model.Users
 		assert.Equal(t, db.Map(rows, &dest), ErrRecordNotFound)
 	})
+	t.Run("should rollback if panic happened during transaction", func(t *testing.T) {
+		ctx, cancel := context.WithCancel(context.Background())
+
+		var user *model.Users
+		err := transaction(db.DB(), ctx, nil, func(tx Tx) error {
+			user = &model.Users{
+				FirstName: null.String{},
+				LastName:  null.String{},
+			}
+			_, err := tx.Insert(user)
+			assert.NoError(t, err)
+
+			cancel()
```

Comment:
As codecov's report telling, cancelation here has no effects to `tx.BeginTx()`, because this callback is called after that. So you must call `cancel()` before calling `transaction()`, maybe?

Reply:
> The provided context is used until the transaction is committed or rolled back. 

Even after the call to `tx.BeginTx()`, if it is canceled before commit or rollback, there should be no problem